File: client/src/types/residentCare.ts
// Resident-care risk asset — safety programs + MVR reviews.

export type ProgramType =
  | 'fall_prevention' | 'infection_control' | 'abuse_prevention'
  | 'emergency_prep' | 'medication_safety' | 'other'
export type ProgramStatus = 'active' | 'inactive'
export type ReviewType = 'hire' | 'annual'
export type MvrStatus = 'clear' | 'flagged' | 'pending'

export interface SafetyProgram {
  id: string
  company_id: string
  program_type: ProgramType
  name: string
  status: ProgramStatus
  last_reviewed_date: string | null
  owner: string | null
  notes: string | null
  created_at: string
}

export interface MvrReview {
  id: string
  company_id: string
  driver_name: string
  employee_id: string | null
  review_type: ReviewType
  review_date: string | null
  status: MvrStatus
  next_due_date: string | null
  notes: string | null
  created_at: string
}

export interface ResidentCareSummary {
  programs: { active: number; total: number; active_types: string[] }
  mvr: { total: number; flagged: number; overdue: number; current: number }
  credentialing: { total: number; expired: number; current_pct: number | null }
}

export const PROGRAM_LABELS: Record<ProgramType, string> = {
  fall_prevention: 'Fall prevention',
  infection_control: 'Infection control',
  abuse_prevention: 'Abuse prevention',
  emergency_prep: 'Emergency preparedness',
  medication_safety: 'Medication safety',
  other: 'Other',
}

export const PROGRAM_TYPES: ProgramType[] = [
  'fall_prevention', 'infection_control', 'abuse_prevention',
  'emergency_prep', 'medication_safety', 'other',
]

File: server/app/matcha/models/insurance/resident_care.py
"""Resident-care risk asset models — safety programs + MVR reviews."""

from datetime import date
from typing import Literal, Optional
from uuid import UUID

from pydantic import BaseModel, Field

ProgramType = Literal[
    "fall_prevention", "infection_control", "abuse_prevention",
    "emergency_prep", "medication_safety", "other",
]
ProgramStatus = Literal["active", "inactive"]
ReviewType = Literal["hire", "annual"]
MvrStatus = Literal["clear", "flagged", "pending"]


class SafetyProgramCreate(BaseModel):
    program_type: ProgramType
    name: str = Field(..., min_length=1, max_length=255)
    status: ProgramStatus = "active"
    last_reviewed_date: Optional[date] = None
    owner: Optional[str] = None
    notes: Optional[str] = None


class SafetyProgramUpdate(BaseModel):
    program_type: Optional[ProgramType] = None
    name: Optional[str] = None
    status: Optional[ProgramStatus] = None
    last_reviewed_date: Optional[date] = None
    owner: Optional[str] = None
    notes: Optional[str] = None


class MvrReviewCreate(BaseModel):
    driver_name: str = Field(..., min_length=1, max_length=255)
    employee_id: Optional[UUID] = None
    review_type: ReviewType = "annual"
    review_date: Optional[date] = None
    status: MvrStatus = "pending"
    next_due_date: Optional[date] = None
    notes: Optional[str] = None


class MvrReviewUpdate(BaseModel):
    driver_name: Optional[str] = None
    review_type: Optional[ReviewType] = None
    review_date: Optional[date] = None
    status: Optional[MvrStatus] = None
    next_due_date: Optional[date] = None
    notes: Optional[str] = None
